src/rm_topk.c

```diff
@@ -0,0 +1,245 @@
+//#include <math.h>    // ceil, log10f
+//#include <strings.h> // strncasecmp
+
+#include "version.h"
+#include "rmutil/util.h"
+
+#include "topk.h"
+#include "rm_topk.h"
+
+#define INNER_ERROR(x)                                                                             \
+    RedisModule_ReplyWithError(ctx, x);                                                            \
+    return REDISMODULE_ERR;
+
+RedisModuleType *TopKType;
+
+static int GetTopKKey(RedisModuleCtx *ctx, RedisModuleString *keyName, TopK **topk, int mode) {
+    // All using this function should call RedisModule_AutoMemory to prevent memory leak
+    RedisModuleKey *key = RedisModule_OpenKey(ctx, keyName, mode);
+    if (RedisModule_KeyType(key) == REDISMODULE_KEYTYPE_EMPTY) {
+        INNER_ERROR("TopK: key does not exist");
```

Comment:
Must call `RedisModule_CloseKey` before returning with an error.

Reply:
I've used RedisModule_AutoMemory. Will change to manual closing.